client/me/purchases/manage-purchase/index.jsx

```diff
@@ -460,20 +447,21 @@ class ManagePurchase extends Component {
 
 export default connect( ( state, props ) => {
 	const purchase = getByPurchaseId( state, props.purchaseId );
-	const selectedSiteId = getSelectedSiteId( state );
+	const selectedSiteId = purchase ? purchase.siteId : null;
```

Comment:
Maybe consider renaming to `purchaseSiteId` to avoid confusion with our pervasive de-facto semantics of what `selectedSiteId` usually denotes. (Same for `selectedSite` below.)

Also, `get( purchase, [ siteId ] )`?

Reply:
I agree that `selectedSite` means something particular in Calypso, and changing the names may help reduce confusion.

I'll look at this just for the `ManagePurchase` and `RemovePurchase` components.

* `selectedSiteId` -> `siteId`
* `selectedSite` -> `site`

This is straightforward for the `ManagePurchase` component, however there

Reply:
9ed78ff9ef4503a7f5f3adabeda6591c501f0b0a